pkg/wpa/passphrase/passphrase_test.go

```diff
@@ -0,0 +1,103 @@
+package passphrase
+
+import (
+	"bytes"
+	"fmt"
+	"testing"
+)
+
+type RunTestCase struct {
+	name  string
+	essid string
+	pass  string
+	out   []byte
+	err   error
+}
+
+var (
+	essidStub     = "stub"
+	shortPass     = "aaaaaaa"                                                          // 7 chars
+	longPass      = "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa" // 64 chars
+	validPass     = "aaaaaaaaaaaaaaaa"                                                 // 16 chars
+	correctOutput = []byte(
+		`network={
+	ssid="stub"
+	#psk="aaaaaaaaaaaaaaaa"
+	psk=e270ba95a72c6d922e902f65dfa23315f7ba43b69debc75167254acd778f2fe9
+}
+`) // indentation matters
+
+	runTestCases = []RunTestCase{
+		{
+			name:  "No essid",
+			essid: "",
+			pass:  validPass,
+			out:   nil,
+			err:   fmt.Errorf("essid cannot be empty"),
+		},
+		{
+			name:  "pass length is less than 8 chars",
+			essid: essidStub,
+			pass:  shortPass,
+			out:   nil,
+			err:   fmt.Errorf("Passphrase must be 8..63 characters"),
+		},
+		{
+			name:  "pass length is more than 63 chars",
+			essid: essidStub,
+			pass:  longPass,
+			out:   nil,
+			err:   fmt.Errorf("Passphrase must be 8..63 characters"),
+		},
+		{
+			name:  "Correct Input",
+			essid: essidStub,
+			pass:  validPass,
+			out:   correctOutput,
+			err:   nil,
+		},
+	}
+)
+
+// Helper function to craft the message
+func craftPrintMsg(err error, out []byte) string {
+	var msg bytes.Buffer
+	msg.WriteString(fmt.Sprintf("Error Status: %v\n", err))
+	msg.WriteString("Output:\n")
+	msg.Write(out)
+	return msg.String()
+}
+
+func outEqualsExp(out []byte, exp []byte) bool {
```

Comment:
not necessary. See:
https://play.golang.org/p/_yh13OpKdvQ
when in doubt, try it out on the playground.

Reply:
Fixed